### PR TITLE
du: replace getopts with clap

### LINKED ISSUE #2120: du: move from getopts to clap
Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/du/src/du.rs

Tests:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_du.rs

As example:
https://github.com/uutils/coreutils/commit/9081e120d6b866c94679d9e29c7bdb17ccccb4d0

### PR DESCRIPTION
**Pull request by syukronrm**
Closes #2120

### CODE REVIEW

File: src/uu/du/src/du.rs [hunk 1]
```diff
@@ -37,6 +38,27 @@ use winapi::um::winbase::GetFileInformationByHandleEx;
...
+    pub const NULL: &str = "0";
+    pub const ALL: &str = "all";
+    pub const APPARENT_SIZE: &str = "apparent-size";
+    pub const BLOCK_SIZE: &str = "B";
+    pub const BYTES: &str = "b";
+    pub const TOTAL: &str = "c";
+    pub const MAX_DEPTH: &str = "d";
+    pub const HUMAN_READABLE: &str = "h";
+    pub const K: &str = "k";
+    pub const L: &str = "l";
```
Comment (sylvestre) [inline src/uu/du/src/du.rs:51]: please rename  the one letter option by something more explicit
it is not obvious when we read options::L

**Comment by sylvestre** [comment]
besides that, looks great :)

File: src/uu/du/src/du.rs [hunk 2]
```diff
@@ -361,126 +384,187 @@ fn convert_size_other(size: u64, _multiplier: u64, block_size: u64) -> String {
     format!("{}", ((size as f64) / (block_size as f64)).ceil())
 }
 
+fn get_usage() -> String {
+    format!(
+        "[OPTION]... [FILE]...
```
Comment (jhscheer) [inline src/uu/du/src/du.rs:389]: there are brackets missing, the output should be
```
du [OPTION]... [FILE]...
du [OPTION]... --files0-from=F
```

Comment (syukronrm) [reply src/uu/du/src/du.rs:389]: Good catch!

File: src/uu/du/src/du.rs [hunk 3]
```diff
@@ -565,17 +654,19 @@ Try '{} --help' for more information.",
...
                         // See: http://linux.die.net/man/2/stat
                         stat.blocks * 512
                     };
-                    if matches.opt_present("time") {
+                    if matches.is_present(options::TIME) {
                         let tm = {
                             let secs = {
-                                match matches.opt_str("time") {
+                                match matches.value_of(options::TIME) {
                                     Some(s) => match &s[..] {
```
Comment (ycd) [inline src/uu/du/src/du.rs:670]: Can you fix the clippy warnings too?
See: https://rust-lang.github.io/rust-clippy/master/#redundant_slicing

File: src/uu/du/src/du.rs [hunk 4]
```diff
@@ -495,42 +579,43 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
             convert_size_m
         } else {
             convert_size_other
         }
     };
     let convert_size = |size| convert_size_fn(size, multiplier, block_size);
 
-    let time_format_str = match matches.opt_str("time-style") {
+    let time_format_str = match matches.value_of("time-style") {
         Some(s) => match &s[..] {
```
Comment (ycd) [inline src/uu/du/src/du.rs:619]: Ditto
See: https://rust-lang.github.io/rust-clippy/master/#redundant_slicing

File: src/uu/du/src/du.rs [hunk 5]
```diff
@@ -361,126 +384,187 @@ fn convert_size_other(size: u64, _multiplier: u64, block_size: u64) -> String {
...
+                    "show times using style STYLE: \
+                    full-iso, long-iso, iso, +FORMAT FORMAT is interpreted like 'date'"
+                )
+        )
+        .arg(
+            Arg::with_name(options::FILE)
+                .hidden(true)
+                .multiple(true)
+        )
+        .get_matches_from(args.clone());
```
Comment (ycd) [inline src/uu/du/src/du.rs:563]: Redundant clone
See: https://rust-lang.github.io/rust-clippy/master/#redundant_clone

**Review by ycd — CHANGES_REQUESTED** [review]
LGTM, it'd be great if you fix the clippy warnings too :)

File: src/uu/du/src/du.rs (hunk 3 above)
Comment (syukronrm) [reply src/uu/du/src/du.rs]: Fixed :+1:

File: src/uu/du/src/du.rs (hunk 4 above)
Comment (syukronrm) [reply src/uu/du/src/du.rs]: Fixed :+1:

File: src/uu/du/src/du.rs (hunk 5 above)
Comment (syukronrm) [reply src/uu/du/src/du.rs]: Fixed :+1:

**Comment by syukronrm** [comment]
Updated.

File: src/uu/du/src/du.rs [hunk 6]
```diff
@@ -361,126 +384,187 @@ fn convert_size_other(size: u64, _multiplier: u64, block_size: u64) -> String {
...
+                .short("s")
+                .long("summarize")
+                .help("display only a total for each argument")
         )
-        .parse(args);
+        .arg(
+            Arg::with_name(options::SI)
+                .long(options::SI)
+                .help("like -h, but use powers of 1000 not 1024")
+        )
```
Comment (sylvestre) [inline src/uu/du/src/du.rs:519]: when you get a chance, could you please open good first issues for these three options? thanks

if people are looking to new bugs to contribute

**Review by sylvestre — APPROVED** [review]
waiting for the CI results

**Comment by sylvestre** [comment]
fails on:

```

---- test_du::test_du_time stdout ----
current_directory_resolved: 
run: /target/i686-unknown-linux-musl/debug/coreutils touch -a -m -t 201505150000 date_test
run: /target/i686-unknown-linux-musl/debug/coreutils du --time date_test
thread 'test_du::test_du_time' panicked at 'Command was expected to succeed.
stdout = 
 stderr = du: error: invalid argument 'modified' for '--time'
    Valid arguments are:
      - 'accessed', 'created', 'modified'
    Try 'du --help' for more information.
', tests/common/util.rs:143:13
```

**Comment by tertsdiepraam** [comment]
The problem is that clap parses this: `du --time date_test` as setting the value for `--time` to `date_test`. This can be fixed by adding `require_equals` and `min_values`:
```rust
.arg(
    Arg::with_name(options::TIME)
        .long(options::TIME)
        .value_name("WORD")
        .require_equals(true)
        .min_values(0)
        .help(
            "show time of the last modification of any file in the \
            directory, or any of its subdirectories.  If WORD is given, show time as WORD instead \
            of modification time: atime, access, use, ctime or status"
        )
)
```

**Comment by syukronrm** [comment]
@tertsdiepraam Thanks! Works well.

**Pull request merged**